packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/models/models.ts

```diff
@@ -825,66 +690,197 @@ export interface _OperationListResult {
   actionType?: ActionType;
 }
 
+export function operationSerializer(input: Operation): unknown {
+  console.log(input);
+  throw new Error("Not implemented");
+}
+
 /** Localized display information for and operation. */
 export interface OperationDisplay {
   /** The localized friendly form of the resource provider name, e.g. "Microsoft Monitoring Insights" or "Microsoft Compute". */
-  readonly provider?: string;
+  provider?: string;
   /** The localized friendly name of the resource type related to this operation. E.g. "Virtual Machines" or "Job Schedule Collections". */
-  readonly resource?: string;
+  resource?: string;
   /** The concise, localized friendly name for the operation; suitable for dropdowns. E.g. "Create or Update Virtual Machine", "Restart Virtual Machine". */
-  readonly operation?: string;
+  operation?: string;
   /** The short, localized friendly description of the operation; suitable for tool tips and detailed views. */
-  readonly description?: string;
-}
-
-/** Known values of {@link Origin} that the service accepts. */
-export enum KnownOrigin {
-  /** user */
-  user = "user",
-  /** system */
-  system = "system",
-  /** user,system */
-  "user,system" = "user,system",
-}
-
-/**
- * The intended executor of the operation; as in Resource Based Access Control (RBAC) and audit logs UX. Default value is "user,system" \
- * {@link KnownOrigin} can be used interchangeably with Origin,
- *  this enum contains the known values that the service supports.
- * ### Known values supported by the service
- * **user** \
- * **system** \
- * **user,system**
- */
-export type Origin = string;
-
-/** Known values of {@link ActionType} that the service accepts. */
-export enum KnownActionType {
-  /** Internal */
-  Internal = "Internal",
-}
-
-/**
- * Extensible enum. Indicates the action type. "Internal" refers to actions that are for internal only APIs. \
- * {@link KnownActionType} can be used interchangeably with ActionType,
- *  this enum contains the known values that the service supports.
- * ### Known values supported by the service
- * **Internal**
- */
-export type ActionType = string;
+  description?: string;
+}
+
+export function operationDisplaySerializer(input: OperationDisplay): unknown {
+  console.log(input);
+  throw new Error("Not implemented");
+}
+
+/** The status of the current operation. */
+export enum ProvisioningStateKnownValues {
+  /** Represents a succeeded operation. */
+  Succeeded = '"Succeeded"',
+  /** Represents a failed operation. */
+  Failed = '"Failed"',
+  /** Represents a canceled operation. */
+  Canceled = '"Canceled"',
+  /** Represents a pending operation. */
+  Provisioning = '"Provisioning"',
+  /** Represents a pending operation. */
+  Updating = '"Updating"',
+  /** Represents an operation under deletion. */
+  Deleting = '"Deleting"',
+  /** Represents an accepted operation. */
+  Accepted = '"Accepted"',
+}
+
+export type ProvisioningState =
```

Comment:
this type is changed from string to union of literal? which should be string as we have enabled experimental extensible enum for this case.